tools/README.md

```diff
@@ -58,6 +67,119 @@ Prints info for block files
 - `<files>...`
    - The block files or directories of block files to print info for
 
-## Running from command line
+### The `record2block` Subcommand
 
-Refer to the [Quickstart](docs/quickstart.md) for a quick guide on how to run the tools CLI.
+Converts a historical record stream files into blocks. This depends on the `block_times.bin` file being present. It can 
+be created by running the other commands `fetchRecordsCsv`, `extractBlockTimes` and `addNewerBlockTimes` in that order. 
+It can also be validated by running the `validateBlockTimes` command. 
+
+This command depends on reading data from public requester pays Google Cloud buckets. To do that it needs you to be 
+authenticated with the Google Cloud SDK. You can authenticate with `gcloud auth application-default login` or 
+`gcloud auth login` see [Google Documentation](https://cloud.google.com/storage/docs/reference/libraries#authentication) 
+for more info.
+
+`Usage: record2block [-s 0] [-e 100] [-j] [-c] [--min-node-account-id=3] [--max-node-account-id=34] [-d <dataDir>] [--block-times=<blockTimesFile>] `
+
+**Options:**
+
+- `-s <blockNumber>` or `--start-block=<blockNumber>`
+   - The first block number to process
+   - Default: 0
+- `-e <blockNumber>` or `--end-block=<blockNumber>`
+   - The last block number to process
+   - Default: 3001
+- `-j` or `--json`
+   - also output blocks as json, useful for debugging and testing
+   - Default: false
+- `-c` or `--cache-enabled`
+   - Use local cache for downloaded content, saves cloud costs and bandwidth when testing
+   - Default: false
+- `--min-node-account-id=<minNodeAccountId>`
+   - the account id of the first node in the network
+   - Default: 3
+- `--max-node-account-id=<maxNodeAccountId>`
+   - the account id of the last node in the network
+   - Default: 34
+- `--data-dir=<dataDir>`
+   - the data directory for output and temporary files
+   - Default: "data"
+- `--block-times=<blockTimesFile>`
+   - Path to the block times ".bin" file.
+   - Default: "data/block_times.bin"
+
+### The `fetchRecordsCsv` Subcommand
+
+Download mirror node record table CSV dump from GCP bucket. The records table on mirror node has a row for every block 
+mirror node knows about. The CSV file is huge 11GB+ in November 2024. This data is important for records to blocks 
+conversion as we have to make sure the block number assigned for a record file matches what mirror node says as the 
+source of truth. 
```

Comment:
Question here, are we saying that the block number in mirror node doesn't match the block number in the record file?

Reply:
Older record files did not have block numbers in them. They were only added maybe 18 mths ago. For a long time block numbers were assigned on mirror node and only existed there.